ls1intum/Artemis
# `General`: Add default programming language to courses

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context



Instructors must select the programming language each time they create a programming exercise.
Generally, a course will use the same programming language from start to finish.
Therefore, the user experience can be improved by choosing a default programming language when creating/editing a course, which will the automatically be pre-set when creating a programming exercise.

### Description


This pull request adds a **new optional attribute** to Courses: `defaultProgrammingLanguage` (String, VARCHAR 255)
On the client, this attribute may be set when creating or updating a Course from the existing list of available programming languages (and an empty default option).
When creating a programming exercise, the programming language will be pre-selected accordingly in the dropdown menu.

> This is one of my first pull requests on this project, so constructive feedback is very welcome.

### Steps for Testing

Prerequisites:
- 1 Instructor

Steps:
1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new Course or select an existing one
4. Choose a default programming language
5. Create a new programming exercise: the corresponding programming language should be pre-selected
7. Make sure all other exercise settings (e.g., template) that depend on the programming language are shown correct

### Review Progress





#### Code Review
#### Manual Tests


### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| course-update.component.ts | 72% | 89% |
| programming-exercise-update.component.ts | 74% | 77% |


### Screenshots



<img width="982" src="https://user-images.githubusercontent.com/102237363/162430675-3220ef81-6941-4e13-b4d9-de7ea06a7ca6.png">
